src/transformers/generation_utils.py

```diff
@@ -841,6 +841,34 @@ def compute_transition_beam_scores(
 
         return transition_scores
 
+    def _validate_model_kwargs(self, model_kwargs: Dict[str, Any]):
+        """Validates model kwargs for generation. Generate argument typos will also be caught here."""
+        # Excludes arguments that are handled before calling any model function
+        if self.config.is_encoder_decoder:
+            for key in ["decoder_input_ids"]:
+                model_kwargs.pop(key, None)
+
+        # Transfo_XL does not use have "attention_mask" as an argument, and it is harmless (it is being passed in the
+        # tests, through, hence this ad hoc exception)
+        if "transfoxl" in str(self).lower():
```

Comment:
Maybe check for `self.config.__class__.__name__` instead here.

Reply:
Think I'd be in favor of displaying the normal "this is not used" error message also for Transfo-XL attention_mask even if we use it in the test -> maybe we should update the tests then instead. If Transfo-XL doesn't have `attention_mask` and the user passes it we should state that it is indeed not used no?

Reply:
(removed this exception, will fix tests correspondingly)